app/models/aid_application.rb

```diff
@@ -429,6 +431,13 @@ def unmet_needs_required
     end
   end
 
+  def zip_code_must_be_in_organization_service_area
+    zip_code_counties = ZipCode.new(zip_code).counties
+    if (organization.county_names & zip_code_counties).empty?
+      errors.add(:zip_code, I18n.t('activerecord.errors.messages.zip_code_outside_organization_service_area', zip_code: zip_code, zip_code_county: zip_code_counties.first))
```

Comment:
This is Rails porcelain plating, but there is a much prettier error object if you use the key name directly for a validation error.

```suggestion
      errors.add(:zip_code, :zip_code_outside_organization_service_area, zip_code: zip_code, zip_code_county: zip_code_counties.first))
```